--- src/master/main.c	2018-04-02 20:39:50.757588253 +0000
+++ src/master/main.c.new	2018-04-03 07:01:01.817479483 +0000
@@ -810,7 +810,11 @@
 		const char **args;
 
 		args = t_new(const char *, 5);
-		args[0] = DOVECOT_CONFIG_BIN_PATH;
+		if (getenv("DOVECOT_BINDIR") != NULL) {
+			argv[0] = t_strconcat(getenv("DOVECOT_BINDIR"), "/doveconf");
+		} else {
+			args[0] = DOVECOT_CONFIG_BIN_PATH;
+		}
 		args[1] = doveconf_arg;
 		args[2] = "-c";
 		args[3] = master_service_get_config_path(master_service);
